src/huggingface_hub/utils/_validators.py

```diff
@@ -40,7 +40,11 @@ class HFValidationError(ValueError):
     """
 
 
-def validate_hf_hub_args(fn: Callable) -> Callable:
+# type hint meaning "function signature not changed by decorator"
+CT = TypeVar("CT")  # callable type
```

Comment:
Fair enough, I renamed it to `CallableT` for more descriptive.

I agree it's hard to use type hints on decorators in Python (see [this discussion](https://github.com/python/mypy/issues/3157) for example). I'm trying to stick to it as much as possible because that's what is sometimes used by IDEs to generate help. Here is a small example in vscode:


![deco](https://user-images.githubusercontent.com/11801849/191267537-165539d3-a0f9-4114-babe-4f37f6a84b89.png)
![deco2](https://user-images.githubusercontent.com/11801849/191267574-64158d3b-3992-46f3-9bae-6a98878e234a.png)